doc: nrf-bm: updates to migration guide

eivindj-nordic opened the pull request:
* Corrections of supported BLE services and libraries
* Explanation of error code use in BM repo.

eivindj-nordic committed:
doc: nrf-bm: updates to migration guide

* Corrections of supported BLE services and libraries
* Explanation of error code use in BM repo.

Signed-off-by: Eivind Jølsgard <eivind.jolsgard@nordicsemi.no>

lemrey commented on doc/nrf-bm/migration/nrf5_bm_migration.rst:
```diff
@@ -255,6 +259,30 @@ This is due to several factors, including:
+***********
+
+nRF5 mainly used the unsigned nRF error scope for error handling, with errors such as ``NRF_ERROR_NO_MEM``, ``NRF_ERROR_FORBIDDEN`` and ``NRF_SUCCESS`` for successful operations.
+In addition the SoftDevice and nrfx provided BLE and nRFX error codes within the same error scope, but with a diffeent base offset.
```
I understand what you want to say here, these were really similar, however they were in effect different errors codes, defined in different files and with different names.

anhmolt committed:
doc: nrf-bm: migration: update links to nrfx migration guides

Be specific and link to both the nrfx-2.11.0-to-3.0.0 and
the nrfx-3.14.0-to-4.0.0 nrfx migration guides.

Signed-off-by: Andreas Moltumyr <andreas.moltumyr@nordicsemi.no>
